packages/service-worker/src/services/settings/handlers/wallet_setSettings.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  DAppRequestHandler,
+  DAppProviderRequest,
+  JsonRpcRequestParams,
+  SettingsState,
+  Languages,
+  CURRENCIES,
+  AnalyticsConsent,
+  ColorTheme,
+  ViewMode,
+  TokensVisibility,
+  CollectiblesVisibility,
+} from '@core/types';
+import { injectable } from 'tsyringe';
+import { SettingsService } from '../SettingsService';
+import { z } from 'zod';
+
+type PartialSettings = Omit<SettingsState, 'customTokens'>;
+type Params = [settings?: PartialSettings];
+
+// Separate return type for the handler API response
+export interface WalletSetSettingsResponse {
+  currency: string;
+  showTokensWithoutBalances: boolean;
+  theme: ColorTheme;
+  tokensVisibility: TokensVisibility;
+  collectiblesVisibility: CollectiblesVisibility;
+  analyticsConsent: AnalyticsConsent;
+  language: Languages;
+  coreAssistant: boolean;
+  preferredView: ViewMode;
+  showTrendingTokens: boolean;
+}
+
+const SettingsSchema = z.object({
+  language: z.nativeEnum(Languages).optional(),
+  currency: z.nativeEnum(CURRENCIES).optional(),
+  analyticsConsent: z.nativeEnum(AnalyticsConsent).optional(),
+  theme: z.enum(['LIGHT', 'DARK', 'SYSTEM']).optional(),
+  preferredView: z.enum(['floating', 'sidebar']).optional(),
+  showTokensWithoutBalances: z.boolean().optional(),
+  coreAssistant: z.boolean().optional(),
+  showTrendingTokens: z.boolean().optional(),
+  tokensVisibility: z
+    .record(z.string(), z.record(z.string(), z.boolean()))
+    .optional(),
+  collectiblesVisibility: z
+    .record(z.string(), z.record(z.string(), z.boolean()))
+    .optional(),
+});
```

Comment:
_Question:_ is an empty object here a valid payload? Since everything is `.optional()`?